### PR TITLE
Spread deploy tiflash to dedicated places

### PR DESCRIPTION
**Pull request by zanmato1984**
### What is changed, added or deleted? (Required)

Delete deploy tiflash doc, and spread its content to several dedicated docs.

Note: `aliases` for the deleted file is added in #3532.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): #3532

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 | TiDB | 8核+ | 16 GB+ | 无特殊要求 | 千兆网卡 | 1（可与 PD 同机器） |
 | PD | 4核+ | 8 GB+ | SAS, 200 GB+ | 千兆网卡 | 1（可与 TiDB 同机器） |
 | TiKV | 8核+ | 32 GB+ | SSD, 200 GB+ | 千兆网卡 | 3 |
+| TiFlash | 32核+ | 64 GB+ | 1 SSD + n HDD, 共 200 GB+ | 千兆网卡 | 1 |
```
Comment (lilin90) [inline hardware-and-software-requirements.md:57]:
```suggestion
| TiFlash | 32 核+ | 64 GB+ | 1 SSD + n HDD，共 200 GB+ | 千兆网卡 | 1 |
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -70,13 +71,18 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 | TiDB | 16核+ | 32 GB+ | SAS | 万兆网卡（2块最佳） | 2 |
```
Comment (lilin90) [inline hardware-and-software-requirements.md:71]:
```suggestion
| TiDB | 16 核+ | 32 GB+ | SAS | 万兆网卡（2 块最佳） | 2 |
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -70,13 +71,18 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
 | PD | 4核+ | 8 GB+ | SSD | 万兆网卡（2块最佳） | 3 |
 | TiKV | 16核+ | 32 GB+ | SSD | 万兆网卡（2块最佳） | 3 |
+| TiFlash | 48核+ | 128 GB+ | 1 NVMe SSD + n HDD | 万兆网卡（2块最佳） | 2 |
 | 监控 | 8核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
```
Comment (lilin90) [inline hardware-and-software-requirements.md:75]:
```suggestion
| PD | 4核+ | 8 GB+ | SSD | 万兆网卡（2块最佳） | 3 |
| TiKV | 16 核+ | 32 GB+ | SSD | 万兆网卡（2 块最佳） | 3 |
| TiFlash | 48 核+ | 128 GB+ | 1 NVMe SSD + n HDD | 万兆网卡（2 块最佳） | 2 |
| 监控 | 8 核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
```

File: tiflash-deployment-topology.md [hunk 4]
```diff
@@ -32,6 +32,14 @@ category: how-to
 
 - TiFlash 具体的参数配置介绍可参考 [TiFlash 参数配置](/tiflash/tiflash-configuration.md)。
 
+### 在已有 TiDB 集群上新增 TiFlash 拓扑
+
+1. 首先确认当前 TiDB 的版本支持 TiFlash，否则需要先升级 TiDB 集群至 4.0 rc 以上版本。
+
+2. 使用 pd-ctl 开启 PD 的 [Placement Rules](/configure-placement-rules.md) 功能，参考 [开启 Placement Rules 特性](/configure-placement-rules.md#开启-placement-rules-特性)。
+
+3. 参考 [扩容 TiFlash 节点](/scale-tidb-using-tiup.md#扩容-tiflash-节点) 章节对 TiFlash 进行部署。
```
Comment (lilin90) [inline tiflash-deployment-topology.md:41]:
```suggestion
2. 使用 pd-ctl 开启 PD 的 [Placement Rules](/configure-placement-rules.md) 功能，参考[开启 Placement Rules 特性](/configure-placement-rules.md#开启-placement-rules-特性)。

3. 参考[扩容 TiFlash 节点](/scale-tidb-using-tiup.md#扩容-tiflash-节点) 章节对 TiFlash 进行部署。
```

File: tiflash/troubleshoot-tiflash.md [hunk 5]
```diff
@@ -38,15 +38,15 @@ category: reference
...
+1. 使用 pd-ctl 检查 PD 的 [Placement Rules](/configure-placement-rules.md) 功能是否开启：
 
     {{< copyable "shell-regular" >}}
 
     ```shell
     echo 'config show replication' | /path/to/pd-ctl -u http://<pd-ip>:<pd-port>
     ```
 
-    预期结果为 `"enable-placement-rules": "true"`。
+    预期结果为 `"enable-placement-rules": "true"`（已开启）。如未开启，具体开启方法参考 [开启 Placement Rules 特性](/configure-placement-rules.md#开启-placement-rules-特性)。
```
Comment (lilin90) [inline tiflash/troubleshoot-tiflash.md:49]:
```suggestion
    预期结果为 `"enable-placement-rules": "true"`（已开启）。如未开启，具体开启方法参考[开启 Placement Rules 特性](/configure-placement-rules.md#开启-placement-rules-特性)。
```

File: hardware-and-software-requirements.md [hunk 6]
```diff
@@ -54,6 +54,7 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 | TiDB | 8核+ | 16 GB+ | 无特殊要求 | 千兆网卡 | 1（可与 PD 同机器） |
 | PD | 4核+ | 8 GB+ | SAS, 200 GB+ | 千兆网卡 | 1（可与 TiDB 同机器） |
 | TiKV | 8核+ | 32 GB+ | SSD, 200 GB+ | 千兆网卡 | 3 |
```
Comment (lilin90) [inline hardware-and-software-requirements.md:56]:
```suggestion
| TiDB | 8 核+ | 16 GB+ | 无特殊要求 | 千兆网卡 | 1（可与 PD 同机器） |
| PD | 4 核+ | 8 GB+ | SAS, 200 GB+ | 千兆网卡 | 1（可与 TiDB 同机器） |
| TiKV | 8 核+ | 32 GB+ | SSD, 200 GB+ | 千兆网卡 | 3 |
```

File: hardware-and-software-requirements.md [hunk 7]
```diff
@@ -51,9 +51,10 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 
 | **组件** | **CPU** | **内存** | **本地存储** | **网络** | **实例数量(最低要求)** |
 | --- | --- | --- | --- | --- | --- |
-| TiDB | 8核+ | 16 GB+ | 无特殊要求 | 千兆网卡 | 1（可与 PD 同机器） |
-| PD | 4核+ | 8 GB+ | SAS, 200 GB+ | 千兆网卡 | 1（可与 TiDB 同机器） |
-| TiKV | 8核+ | 32 GB+ | SSD, 200 GB+ | 千兆网卡 | 3 |
+| TiDB | 8 核+ | 16 GB+ | 无特殊要求 | 千兆网卡 | 1（可与 PD 同机器） |
+| PD | 4 核+ | 8 GB+ | SAS, 200 GB+ | 千兆网卡 | 1（可与 TiDB 同机器） |
+| TiKV | 8 核+ | 32 GB+ | SSD, 200 GB+ | 千兆网卡 | 3 |
+| TiFlash | 32 核+ | 64 GB+ | 1 SSD + n HDD，共 200 GB+ | 千兆网卡 | 1 |
```
Comment (flowbehappy) [inline hardware-and-software-requirements.md:57]: Currently we don't suggest HDD any more.
```suggestion
| TiFlash | 32 核+ | 64 GB+ | SSD, 200 GB+ | 千兆网卡 | 1 |
```

File: hardware-and-software-requirements.md [hunk 8]
```diff
@@ -67,16 +68,21 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+| TiFlash | 48 核+ | 128 GB+ | 1 NVMe SSD + n HDD | 万兆网卡（2 块最佳） | 2 |
+| 监控 | 8 核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
 
 > **注意：**
 >
 > - 生产环境中的 TiDB 和 PD 可以部署和运行在同服务器上，如对性能和可靠性有更高的要求，应尽可能分开部署。
 > - 生产环境强烈推荐使用更高的配置。
 > - TiKV 硬盘大小配置建议 PCI-E SSD 不超过 2 TB，普通 SSD 不超过 1.5 TB。
+> - TiFlash 支持 [多盘部署](/tiflash/tiflash-configuration.md#多盘部署)，所以无需使用 RAID。
+> - TiFlash 推荐用一个 SSD 来缓冲 TiKV 同步数据的实时写入，该盘性能不低于 TiKV 所使用的硬盘，容量建议不小于总容量的 10%，否则它可能成为这个节点的能承载的数据量的瓶颈。而其他硬盘可以选择部署多块 HDD 或者 SSD，更好的硬盘会带来更好的性能。
```
Comment (flowbehappy) [inline hardware-and-software-requirements.md:83]:
```suggestion
> - TiFlash 数据目录的第一个磁盘推荐用高性能 SSD 来缓冲 TiKV 同步数据的实时写入，该盘性能应不低于 TiKV 所使用的磁盘，比如 NVMe SSD。并且该磁盘容量建议不小于总容量的 10%，否则它可能成为这个节点的能承载的数据量的瓶颈。而其他磁盘可以根据需求部署多块普通 SSD，当然更好的 NVMe SSD 硬盘会带来更好的性能。
```

File: hardware-and-software-requirements.md [hunk 9]
```diff
@@ -67,16 +68,21 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 
 | **组件** | **CPU** | **内存** | **硬盘类型** | **网络** | **实例数量(最低要求)** |
 | --- | --- | --- | --- | --- | --- |
-| TiDB | 16核+ | 32 GB+ | SAS | 万兆网卡（2块最佳） | 2 |
+| TiDB | 16 核+ | 32 GB+ | SAS | 万兆网卡（2 块最佳） | 2 |
 | PD | 4核+ | 8 GB+ | SSD | 万兆网卡（2块最佳） | 3 |
-| TiKV | 16核+ | 32 GB+ | SSD | 万兆网卡（2块最佳） | 3 |
-| 监控 | 8核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
+| TiKV | 16 核+ | 32 GB+ | SSD | 万兆网卡（2 块最佳） | 3 |
+| TiFlash | 48 核+ | 128 GB+ | 1 NVMe SSD + n HDD | 万兆网卡（2 块最佳） | 2 |
```
Comment (flowbehappy) [inline hardware-and-software-requirements.md:74]:
```suggestion
| TiFlash | 48 核+ | 128 GB+ | 1 or more SSDs | 万兆网卡（2 块最佳） | 2 |
```

File: hardware-and-software-requirements.md [hunk 10]
```diff
@@ -67,16 +68,21 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+| TiFlash | 48 核+ | 128 GB+ | 1 or more SSDs | 万兆网卡（2 块最佳） | 2 |
+| 监控 | 8 核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
 
 > **注意：**
 >
 > - 生产环境中的 TiDB 和 PD 可以部署和运行在同服务器上，如对性能和可靠性有更高的要求，应尽可能分开部署。
 > - 生产环境强烈推荐使用更高的配置。
 > - TiKV 硬盘大小配置建议 PCI-E SSD 不超过 2 TB，普通 SSD 不超过 1.5 TB。
+> - TiFlash 支持 [多盘部署](/tiflash/tiflash-configuration.md#多盘部署)，所以无需使用 RAID。
+> - TiFlash 数据目录的第一块磁盘推荐用高性能 SSD 来缓冲 TiKV 同步数据的实时写入，该盘性能应不低于 TiKV 所使用的磁盘，比如 NVMe SSD。并且该磁盘容量建议不小于总容量的 10%，否则它可能成为这个节点的能承载的数据量的瓶颈。而其他磁盘可以根据需求部署多块普通 SSD，当然更好的 NVMe SSD 硬盘会带来更好的性能。
```
Comment (flowbehappy) [inline hardware-and-software-requirements.md:83]:
```suggestion
> - TiFlash 数据目录的第一块磁盘推荐用高性能 SSD 来缓冲 TiKV 同步数据的实时写入，该盘性能应不低于 TiKV 所使用的磁盘，比如 PCI-E SSD。并且该磁盘容量建议不小于总容量的 10%，否则它可能成为这个节点的能承载的数据量的瓶颈。而其他磁盘可以根据需求部署多块普通 SSD，当然更好的 PCI-E SSD 硬盘会带来更好的性能。
```

File: hardware-and-software-requirements.md [hunk 11]
```diff
@@ -67,16 +68,21 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+| TiKV | 16 核+ | 32 GB+ | SSD | 万兆网卡（2 块最佳） | 3 |
+| TiFlash | 48 核+ | 128 GB+ | 1 or more SSDs | 万兆网卡（2 块最佳） | 2 |
+| 监控 | 8 核+ | 16 GB+ | SAS | 千兆网卡 | 1 |
 
 > **注意：**
 >
 > - 生产环境中的 TiDB 和 PD 可以部署和运行在同服务器上，如对性能和可靠性有更高的要求，应尽可能分开部署。
 > - 生产环境强烈推荐使用更高的配置。
 > - TiKV 硬盘大小配置建议 PCI-E SSD 不超过 2 TB，普通 SSD 不超过 1.5 TB。
+> - TiFlash 支持 [多盘部署](/tiflash/tiflash-configuration.md#多盘部署)，所以无需使用 RAID。
```
Comment (lilin90) [inline hardware-and-software-requirements.md:82]:
```suggestion
> - TiFlash 支持[多盘部署](/tiflash/tiflash-configuration.md#多盘部署)，所以无需使用 RAID。
```

**Comment by flowbehappy** [comment]
LGTM

File: tiflash-deployment-topology.md [hunk 12]
```diff
@@ -32,6 +32,14 @@ category: how-to
 
 - TiFlash 具体的参数配置介绍可参考 [TiFlash 参数配置](/tiflash/tiflash-configuration.md)。
 
+### 在已有 TiDB 集群上新增 TiFlash 拓扑
```
Comment (kissmydb) [inline tiflash-deployment-topology.md:35]: 这部分内容不需要了吧，在扩容里都包括了。如果担心用户不知道该看扩容，应该加在部署文档（production-deployment-using-tiup.md）中的 tiflash 部分，而不应该是拓扑文档呢~

Comment (zanmato1984) [reply tiflash-deployment-topology.md:35]: 这部分内容留着主要是考虑需要告知用户必须手动开启placement rule。

放在这个地方主要是考虑这个文档专门用来介绍tiflash拓扑的，在已有tidb上添加tiflash也是要编辑拓扑的。而部署文档中没有单独提及tiflash，放了一个“在已有tidb上添加tiflash”显得突兀。

Comment (yikeke) [reply tiflash-deployment-topology.md:35]: 这部分内容留着主要是考虑需要告知用户必须手动开启placement rule。
-------
tiflash-deployment-topology.md 这篇文章 line 29 不是提醒用户需要「将配置模板中 `replication.enable-placement-rules` 设置为 `true`，以开启 PD 的 [Placement Rules](/configure-placement-rules.md) 功能」了吗？ @zanmato1984

Comment (zanmato1984) [reply tiflash-deployment-topology.md:35]: 就是这个地方不同。新集群可以通过配置文件直接打开，老集群必须通过pd ctl打开。

Comment (kissmydb) [reply tiflash-deployment-topology.md:35]: 如果是考虑这个参数，在扩容文档里已经有介绍了
https://pingcap.com/docs-cn/stable/scale-tidb-using-tiup/#%E6%89%A9%E5%AE%B9-tiflash-%E8%8A%82%E7%82%B9

**Comment by zanmato1984** [comment]
@kissmydb PTAL

**Comment by ran-huang** [comment]
Aligned hardware-and-software-requirements.md in https://github.com/pingcap/docs/pull/2730

**Comment by yikeke** [comment]
> Aligned hardware-and-software-requirements.md in [pingcap/docs#2730](https://github.com/pingcap/docs/pull/2730)

The rest of changes are aligned in https://github.com/pingcap/docs/pull/2749

**Pull request merged**